### ISSUE TITLE
Training VoiceCraft with IndicVoices-R

### ISSUE THREAD

**Issue by meets2tarun** [AI4Bharat/IndicVoices-R#5]
Hi,

Thanks for this outstanding work and for achieving this significant milestone in supporting Indian languages.

I would like to train [voicecraft](https://github.com/jasonppy/VoiceCraft) with all the languages in [IndicVoices-R](https://ai4bharat.iitm.ac.in/datasets/IndicVoices-R). I would be starting with the Hindi dataset with the voice-cloning functionality. 

So far, to understand it, I tried to reproduce the training with the Gigaspeech XL dataset on the model. 

It would be great if I get some help from @AshwinSankar17. 

Thanks 
Tarun

**Comment by AshwinSankar17** [comment]
The first step would be to extend the vocab to your grapheme sets if you do not want to use phonemes. You can use [this](https://gist.github.com/AshwinSankar17/6b5b20d88de4e65a3694b14214177f4f) script to do that. Alternatively, you can use the espeak tokenizer or the tokenizer from [xphonebert](https://huggingface.co/vinai/xphonebert-base) if you want to continue using phonemes.

Next you need to make changes to phonemize_encodec_encode_hf.py to load huggingface dataset from json instead of pulling from the web. You may also need to change the tokenization approach here depending on your previous step.

Next in steps/trainer.py you need to extend the embedding table from the old embedding table (to somewhat benefit from the pretraining). 
`
old_token_lim = sd["text_embedding.word_embeddings.weight"].size(0) - 1
old_embeddings = sd["text_embedding.word_embeddings.weight"][:old_token_lim,]
mean = torch.mean(old_embeddings, dim=0).float()
std = torch.std(old_embeddings, dim=0).float()

new_embeddings = torch.from_numpy(
                np.random.normal(
                    loc=mean.numpy(), 
                    scale=std.numpy(), 
                    size=(self.args.text_vocab_size - old_token_lim, self.args.d_model)
                )
            )


            
            sd["text_embedding.word_embeddings.weight"] = torch.cat(
                    (
                        sd["text_embedding.word_embeddings.weight"][:old_token_lim,], 
                        new_embeddings, 
                        sd["text_embedding.word_embeddings.weight"][old_token_lim:,]
                    )
                )
            model.load_state_dict(sd, strict=False)
`

**Comment by AshwinSankar17** [comment]
Closing this if there are no further questions.
